From 4c3c57ad9ce35f6a96f7b360b0272065f1957a51 Mon Sep 17 00:00:00 2001
From: Tim Murray <timmurray@google.com>
Date: Sat, 11 Jul 2020 01:19:07 +0000
Subject: [PATCH 8/8] [android11-qpr1-release] gralloc: Bump allocator priority
 to 100.

 * The allocator can cause stalls elsewhere in the
   system if it's running at prio 120. Bump it to 100,
   the highest non-RT priority.

Change-Id: I59157448f422f8ce9a50059c2baedf1d201a414c
---
 gralloc/service.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/gralloc/service.cpp b/gralloc/service.cpp
index 2b890ffdc..f66ec0503 100644
--- a/gralloc/service.cpp
+++ b/gralloc/service.cpp
@@ -40,6 +40,7 @@ int main(int, char **) {
       new vendor::qti::hardware::display::allocator::V3_0::implementation::QtiAllocator();
 
   configureRpcThreadpool(4, true /*callerWillJoin*/);
+  android::hardware::setMinSchedulerPolicy(service3, SCHED_NORMAL, -20);
   if (service3->registerAsService() != android::OK) {
     ALOGE("Cannot register QTI Allocator 3 service");
     return -EINVAL;
@@ -49,6 +50,7 @@ int main(int, char **) {
 #ifdef TARGET_USES_GRALLOC4
   android::sp<IQtiAllocator4> service4 =
       new vendor::qti::hardware::display::allocator::V4_0::implementation::QtiAllocator();
+  android::hardware::setMinSchedulerPolicy(service4, SCHED_NORMAL, -20);
   if (service4->registerAsService() != android::OK) {
     ALOGE("Cannot register QTI Allocator 4 service");
     return -EINVAL;
-- 
2.34.1

